express-notes-api/index.js

```diff
@@ -0,0 +1,124 @@
+/* eslint-disable no-unused-vars */
+
+const express = require('express');
+const app = express();
+
+const jsonFile = require('./data.json');
+const fs = require('fs');
+
+const notes = [];
+const nextId = jsonFile.nextId;
+
+for (const note in jsonFile.notes) {
+  notes.push(jsonFile.notes[note]);
+}
+
+app.get('/api/notes', (req, res) => {
+  res.json(notes);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const parsedId = parseInt(req.params.id);
+
+  if (parsedId < 0 || isNaN(parsedId)) {
+    res.status(400).json({
+      error: 'id must be a positive integer'
+    });
+  } else {
+    const indexNum = notes.findIndex(note => note.id === parsedId);
+    if (indexNum === -1) {
+      res.status(404).json({
+        error: `cannot find note with id ${parsedId}`
+      });
+    } else {
+      res.json(notes[indexNum]);
+    }
+  }
+});
+
+app.use(express.json());
+
+app.post('/api/notes', (req, res) => {
+  const note = req.body;
+  if (JSON.stringify(note) === '{}') {
+    res.status(400).json({
+      error: 'content is a required field'
+    });
+  } else {
```

Comment:
What if the note is `{ hello: 'evelyn' }`? You can check the `typeof note.content` instead.